File: PDOO/NapakalakiRuby.ruby/lib/treasure.rb
#encoding: utf-8

require_relative 'treasure_kind'

module NapakalakiGame

  class Treasure
    
    private
    
    def initialize(n, bonus, t)
      @name = n
      @bonus = bonus
      @type = t
    end
    
    public

    def getName
      @name
    end

    def getBonus
      @bonus
    end

    def getType
      @type
    end

    def to_s
      "\n\t\tTesoro: #{getName} \n\t\tTipo: #{@type} \n\t\tBonus: #{@bonus}\n"
    end

  end
end

File: PDOO/Napakalaki.java/src/NapakalakiGame/Treasure.java
package NapakalakiGame;

/**
 *
 * @author Andres Arco Lopez y Miguel Angel Cantarero Lopez
 */
public class Treasure {
    
    private String name;
    
    private int bonus;
    
    private TreasureKind type;
    
    public Treasure(String n, int bonus, TreasureKind t){       
        this.name = n;
        this.bonus = bonus;
        this.type = t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    public String toString(){
        return "\n\n\t\tTesoro " + this.name + "\n\t\tTipo: " + getType() + "\n\t\tBonus: " + getBonus();
    }
}
